contracts/src/tests/test_mcms.cairo

```diff
@@ -0,0 +1,1023 @@
+use core::array::{SpanTrait, ArrayTrait};
```

Comment:
any unit tests for the encoding and merkle proof validation util functions?

a full e2e test from set config -> root -> execute would be great to validate as well

Reply:
definitely, the plan was to add more tests in follow up PRs. i thought there was enough changes already for one PR.

Reply:
For features introduced in a single PR, yep this is plenty. However tests (especially unit tests) are a way to validate the correctness of these features as you cant really tell just by reading code when reviewing PRs. Some implementations, e.g. the encoding of the `data` field I mentioned above, I wouldn't know if it worked correctly or not, so seeing tests for such components helps build confidence for PR approvals and reduces the number of bugfix prs later on.

Reply:
yeah that works -- i can keep this PR open for all the tests then